### ISSUE TITLE
feat: Rich Presence (RA) on emulator

### ISSUE THREAD

**Issue by falsepopsky** [flyinghead/flycast#1637]
# Request

Support Rich Presence (RetroAchievements) on Flycast (NO DISCORD). This feature should likely be included in the "Achievements" section.
Several other emulators already support Rich Presence, such as Dolphin, PCSX2, etc.

Riche presence preview on emulators:

**Dolphin**

![Dolphin](https://github.com/user-attachments/assets/ede4169f-31af-4f9a-9c1c-31bcffd1fc84)

**PCSX2**

![PCSX2](https://github.com/user-attachments/assets/c5f4b5e2-2952-4ded-bb74-097fcb7e5d7b)

---

TLDR: Support the Rich Presence in the emulator (NO DISCORD)

**Comment by flyinghead** [comment]
Rich Presence is already supported.
![image](https://github.com/user-attachments/assets/405b34b9-104d-467e-95ff-e8c577f2d1e8)
I don't see the point of displaying it in Flycast since I assume the user knows which game he is playing.

**Comment by falsepopsky** [comment]
Hey thanks for the quick response, the problem is that there are some games (or achievements) that specifically track items, health, and other stuff. A quick dummy (fake) example would be with that game "102 ..." where you need to collect all "secret bones" in "x" level. However, the game doesn't have a way to show you the bones that you've collected until you finish the level. So, to keep an eye on your progress, you can quickly check the Rich Presence feature instead of relying on the website that updates after 2 minutes. 

You can see a more verbose example that I put in the preview on PCSX2.
